### PR TITLE
Support multiple tokens

### PR DESCRIPTION
**Pull request by mach-kernel**
Hey there,

I don't know if this is a very common use case but this is pretty useful to me and doesn't really break anything that is already there. 

Thanks!

### CODE REVIEW

**Comment by jswetzen** [comment]
Hey, thanks for contributing! I don't have any use for it personally, but supporting multiple tokens is a good addition. Your "oops" commit seems to be added to the wrong branch though, so it doesn't show up. I'll make a review with some comments too.

File: update_entrydns.sh [hunk 1]
```diff
@@ -1,12 +1,28 @@
...
+  curl -k -X PUT -d "ip=${IPADDRESS}" "https://entrydns.net/records/modify/$token"  
+}
+
+IPADDRESS=`curl -s icanhazip.com`
+if [[ "${IPADDRESS}" != "$(cat /current_ip)" ]]; then
+
+  if [[ "${AUTH_TOKENS}" ]]; then
+    if [ -n "${AUTH_TOKENS}" ]; then
+        declare part
+        while read -d "," part; do
```
Comment (jswetzen) [inline update_entrydns.sh]: Wouldn't this work on a single token too (no commas)? Then you don't have to treat them as different cases. Does the while loop work without bash?

File: Dockerfile [hunk 2]
```diff
@@ -1,7 +1,7 @@
 FROM alpine:3.5
 MAINTAINER Johan Swetzén <johan@swetzen.com>
 
-RUN apk add --no-cache curl
+RUN apk add --no-cache curl bash
```
Comment (jswetzen) [inline Dockerfile:4]: Is bash needed? (See further comments)

File: update_entrydns.sh [hunk 3]
```diff
@@ -1,12 +1,28 @@
-#!/bin/sh
+#!/bin/bash
 
-[ -z "${AUTH_TOKEN}" ] && { echo "=> AUTH_TOKEN cannot be empty" && exit 1; }
+([ -z "${AUTH_TOKEN}" ] && [ -z "${AUTH_TOKENS}" ]) && { echo "=> AUTH_TOKEN(S) cannot be empty" && exit 1; }
```
Comment (jswetzen) [inline update_entrydns.sh]: Is this the part that needs bash? Use only "AUTH_TOKEN" and it's not needed any more.

**Review by jswetzen — CHANGES_REQUESTED** [review]

**Comment by mach-kernel** [comment]
@jswetzen I tried with `sh` on OS X and was successful in getting the script to run, but not from the Alpine image (so I installed `bash`). I saw your change requests and clean this up -- I should have done so before opening the PR since my solution was very "quick rig up" to replace an internal script we had. 

Thanks! I'll get to it 🚀

**Comment by jswetzen** [comment]
@mach-kernel Ah yes, different `sh` behaviour makes development difficult. I became a bit obsessed with minimal dependencies when I found Alphine, but `bash` is absolutely fine. 👌
If you can handle the list and single token as the same variable I'll merge it right away!

**Comment by jswetzen** [comment]
Have you made any progress on this?

**Comment by mach-kernel** [comment]
@jswetzen it's been a busy few weeks, I'll knock this out today!

**Comment by mach-kernel** [comment]
@jswetzen Today actually meant in 4 days, but there it is 😸 . We do however need `bash` for consistent tokenization support. I'm with it only on reason of opting out from a code golf adventure on finding the best way to do it with `sh` 😆

**Comment by mach-kernel** [comment]
@jswetzen any time to look these over?

**Comment by jswetzen** [comment]
I had a chance to sit down and give it a go myself. I couldn't find a way to update your pull request, but could you have a look at PR #3?

**Comment by jswetzen** [comment]
Sorry for not accepting your solution @mach-kernel, but I added ShellCheck linting and saw that there were some problems with my original script. Then I found that a `for` loop would do the tokenization fine if the Internal Field Separator was changed, so it made for a simple `sh` solution. I'm thankful for your contribution and hope you will find that this image works well for your use case too now.

**Comment by mach-kernel** [comment]
That's awesome! No worries at all.

**Pull request closed without merging**